scapy/layers/dot11.py

```diff
@@ -381,7 +374,7 @@ class Dot11QoS(Packet):
 
     def guess_payload_class(self, payload):
         if isinstance(self.underlayer, Dot11):
-            if self.underlayer.FCfield & 0x40:
+            if self.underlayer.FCfield.protected:
                 return Dot11WEP
```

Comment:
In `Dot11`, you now return a `Dot11Encrypted` instead of `Dot11WEP` for this same condition. Is it wanted to keep `Dot11WEP` here?

Reply:
You are right, I missed that. `Dot11Encrypted` now has a dispatch_hook to select the matching encryption